### PR TITLE
PE-35 | Implemented New Question Pages

### PR DESCRIPTION
**Pull request by tonynguyen98**
**What I did:**

- Created Screens to allow a user to create new questions posts.
- Create some components for things we might use often.

**How To Test :**

- Clone repo and run `flutter pub get`.
- Navigate from the side menu to get to the "Create a New Question Page".
- Select a category from the 4.
- Check that the selector on top will switch between "Text" "Photo" and "Video"
- Return to the "Text" tab and fill out a Question Title and Question Body.
-  Click Submit and watch you debug console, you should see the values you've entered: 
image
- You should then see a confirmation screen, make sure that you can see your title and body here again.

### CODE REVIEW

File: lib/components/eureka_segmented_control.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+/// This is a custom made Segmented Control made specifically for Project
+/// Eureka. Please use this Widget when implementing a segmented control of
+/// some type to prevent us from repeating ourselves.
+class EurekaSegmenetedControl extends StatefulWidget {
+  final int segmentedControlCount;
+  final List<String> segmentedControlLabels;
+  final int segmentedControlValue;
+  final ValueChanged<dynamic> setState;
+
```
Comment (frlzjosh) [inline lib/components/eureka_segmented_control.dart]: consider updating your command to the following 
```
/*
 * comment1
 * comment2
 * /
```

Comment (frlzjosh) [reply lib/components/eureka_segmented_control.dart]: Disregard this comment, I just cited this document... this is diff than what I'm used to in java and python https://dart.dev/guides/language/effective-dart/documentation

File: lib/components/eureka_rounded_button.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+
+/// This is a custom made Rounded Button made specifically for Project
+/// Eureka. Please use this Widget when implementing and big buttons so
```
Comment (ayeejane) [inline lib/components/eureka_rounded_button.dart]: implementing big buttons?

File: lib/components/eureka_text_form_field.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// This is a custom made TextForm made specifically for Project
+/// Eureka. Please use this Widget when any text form fields.
```
Comment (ayeejane) [inline lib/components/eureka_text_form_field.dart]: when using any text form fields

File: lib/screens/new_question_screens/new_question_form.dart [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+  Column _textForm() {
+    return Column(
+      children: <Widget>[
+        EurekaTextFormField(
+          labelText: 'Question Title',
+          errValidatorMsg: 'Question title is required.',
+          regExp: _nameRegExp,
+          onSaved: (value) => _questionTitle = value.trim(),
+        ),
```
Comment (ayeejane) [inline lib/screens/new_question_screens/new_question_form.dart]: Check if the user inputs just a single letter for the `Question Title`. It should also be invalid.

**Review by ayeejane — CHANGES_REQUESTED** [review]

File: lib/components/eureka_rounded_button.dart (hunk 2 above)
Comment (tonynguyen98) [reply lib/components/eureka_rounded_button.dart:4]: I was thinking, whenever we needed a button at the bottom the screen that is similar in size and shape. You can find something like this in profile onboarding and create a new question.

**Review by ayeejane — APPROVED** [review]
Looks good to me!

**Review by davidsquines — APPROVED** [review]
Works as intended and codebase is clean

File: lib/screens/new_question_screens/new_question_confirmation.dart [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                height: 40.0,
+              ),
+              _questionCornfirmationTextStyling(
+                  widget.questionModel.description),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
```
Comment (dominikyz) [inline lib/screens/new_question_screens/new_question_confirmation.dart]: Right now, when question is submitted,  we have 1 gray container that shows both title of question and question itself. May i suggest little correction so it's easier to distinguish where title ends :
easy one - make title text bold and maybe one font-size bigger that the question's text
harder one - display question submission in two separate containers (smaller one on top for title, and larger one on the bottom for question itself)

**Review by dominikyz — COMMENTED** [review]
Code looks awesome, easy to read. Just have one suggestion for question submit page.
Also if user starts to type question, then goes to photo/video upload, and goes back to typing,  typed text is deleted. Should we have temp storage for that? Just a suggestion tho :)  Works as described otherwise

**Pull request merged**